pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -132,12 +125,18 @@ func (c *CloudProvider) create(ctx context.Context, constraints *v1alpha5.Constr
 	return errs
 }
 
+// Despite accepting a Constraints struct, note that it does not utilize Requirements at all and instead
+// returns all available InstanceTypes.
 func (c *CloudProvider) GetInstanceTypes(ctx context.Context, constraints *v1alpha5.Constraints) ([]cloudprovider.InstanceType, error) {
 	vendorConstraints, err := v1alpha1.Deserialize(constraints)
 	if err != nil {
 		return nil, apis.ErrGeneric(err.Error())
 	}
-	return c.instanceTypeProvider.Get(ctx, vendorConstraints)
+	allInstanceTypes, err := c.instanceTypeProvider.Get(ctx, vendorConstraints)
```

Comment:
consider simplifying to:
```suggestion
	instanceTypes, err := c.instanceTypeProvider.Get(ctx, vendorConstraints)
```
"all" can be inferred from context.